src/table/settings.ini

```diff
@@ -3721,21 +3721,75 @@ cat      = SC_BASIC
 base     = GameSettings
 var      = game_creation.min_river_length
 type     = SLE_UINT8
+guiflags = SGF_NEWGAME_ONLY | SGF_SCENEDIT_TOO
 from     = SLV_163
 def      = 16
 min      = 2
 max      = 255
 cat      = SC_EXPERT
+str      = STR_CONFIG_SETTING_RIVER_MIN_LENGTH
+strhelp  = STR_CONFIG_SETTING_RIVER_MIN_LENGTH_HELPTEXT
+strval   = STR_JUST_COMMA
 
 [SDT_VAR]
 base     = GameSettings
 var      = game_creation.river_route_random
 type     = SLE_UINT8
+guiflags = SGF_NEWGAME_ONLY | SGF_SCENEDIT_TOO
 from     = SLV_163
 def      = 5
 min      = 1
 max      = 255
 cat      = SC_EXPERT
+str      = STR_CONFIG_SETTING_RIVER_ROUTE_RANDOM
+strhelp  = STR_CONFIG_SETTING_RIVER_ROUTE_RANDOM_HELPTEXT
+strval   = STR_JUST_COMMA
+
+[SDT_BOOL]
+base     = GameSettings
+var      = game_creation.rivers_top_of_hill
+guiflags = SGF_NEWGAME_ONLY | SGF_SCENEDIT_TOO
+def      = true
+cat      = SC_EXPERT
+str      = STR_CONFIG_SETTING_RIVERS_TOP_OF_HILL
+strhelp  = STR_CONFIG_SETTING_RIVERS_TOP_OF_HILL_HELPTEXT
+
+[SDT_VAR]
+base     = GameSettings
+var      = game_creation.river_tropics_width
+type     = SLE_UINT8
+guiflags = SGF_NEWGAME_ONLY | SGF_SCENEDIT_TOO
+def      = 5
+min      = 1
+max      = 12
+interval = 1
+str      = STR_CONFIG_SETTING_RIVER_TROPICS_WIDTH
+strhelp  = STR_CONFIG_SETTING_RIVER_TROPICS_WIDTH_HELPTEXT
+strval   = STR_JUST_COMMA
+cat      = SC_BASIC
+patxname = ""rainforest_line_height.game_creation.rainforest_line_height""
+
+[SDT_VAR]
+base     = GameSettings
+var      = game_creation.lake_size
+type     = SLE_UINT8
+guiflags = SGF_NEWGAME_ONLY | SGF_SCENEDIT_TOO
+def      = 3
+min      = 1
+max      = 255
```

Comment:
Would it be worth constraining the upper limit a bit?
Diameter 255 lakes seems potentially overkill.

Reply:
I'm not against it. I play with 50 personally, does 100 seem like a reasonable compromise?

Reply:
```suggestion
max      = 100
```

Reply:
That should be fine